### PR TITLE
DJAN-11 Unattended security upgrades

### PR DESCRIPTION
**Pull request by dpoirier**
Add a Salt state 'unattended_upgrades' that will enable
weekly automatic package updates for security fixes.

### CODE REVIEW

File: unattended_upgrades/50unattended-upgrades.j2 [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// 'mailx' must be installed. E.g. "user@example.com"
+Unattended-Upgrade::Mail "{{ admin_email }}";
+
+// Set this value to "true" to get emails only on errors. Default
+// is to always send a mail if Unattended-Upgrade::Mail is set
+Unattended-Upgrade::MailOnlyOnError "false";
+
+// Do automatic removal of new unused dependencies after the upgrade
+// (equivalent to apt-get autoremove)
+Unattended-Upgrade::Remove-Unused-Dependencies "true";
```
Comment (vkurup) [inline unattended_upgrades/50unattended-upgrades.j2]: I've rarely had `apt-get autoremove` a package that i didn't want removed. It's only occured on my desktop, and it was probably due to user-error, but just wondering if we should err on the side of conservatism and keep this "false"?

File: unattended_upgrades/init.sls [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - update-notifier-common
+      - mailutils
+
+install_20auto_upgrades_file:
+  file.managed:
+    - name: /etc/apt/apt.conf.d/20auto-upgrades
+    - source: salt://unattended_upgrades/20auto-upgrades
+    - user: root
+    - group: root
+    - mode: 444
```
Comment (vkurup) [inline unattended_upgrades/init.sls]: Looks like the other files in this directory are mode 644. If it's 444, then root couldn't edit or replace the file, which might make future deployments have issues?

**Comment by vkurup** [comment]
Does this state need to be added to `top.sls`, or are you planning that separately (perhaps with docs, I guess?)

Aside from the minor comments above, looks good to me.

**Comment by dpoirier** [comment]
Updated.

After this, I'll do a new margarita release that includes it, then update django-project-template to add it to `top.sls` and use the new release of margarita.

**Comment by vkurup** [comment]
@mlavin recommends turning auto-remove back on, to avoid having old kernels fill up `/boot`, so that makes sense to me.

:+1: after that. Perhaps we can add a note to the release notes informing upgraders to update their `top.sls` when they upgrade to this release.

**Comment by vkurup** [comment]
:shipit:

**Pull request merged**